crates/astria-sequencer-utils/src/genesis_parser.rs

```diff
@@ -60,11 +64,12 @@ impl GenesisParser {
     }
 }
 
-fn insert_app_state(dst: &mut Value, app_state: &Value) {
+fn insert_app_state_and_chain_id(dst: &mut Value, app_state: &Value, chain_id: String) {
```

Comment:
This doesn't really scale IMO. Better to replace this by
```rust
dst["app_state"] = json!(source_genesis_data);
dst["chain_id"] = json!(chain_id);
```

Reply:
like replace the whole function? that is harder for unit tests :/

Reply:
Yeah, this should be rewritten so that not everything is stuffed into a single function.